src/pages/index.js

```diff
@@ -44,31 +84,113 @@ function openProfilePopup() {
 
 function changeProfileInfo(event, data) {
   user.setUserInfo(data);
-  editPopup.close();
+  api
+    .postUserInfo(data)
+    .then(() => {
+      editPopup.close();
+      toggleButtonText(editPopup.submitButton);
```

Comment:
`      toggleButtonText(editPopup.submitButton);
`

You need to start the loading effect **before** the request to show `Saving...` in the button